hal/architecture/AVR/MyHwMegaAVR.cpp

```diff
@@ -0,0 +1,433 @@
+				// Pause a short while
+				delay(seed % 10);
+				timeout = millis() + 20;
+			}
+		}
+	}
+	randomSeed(seed);
+}
+
+bool hwUniqueID(unique_id_t *uniqueID)
+{
+	// padding
+	(void)memset(uniqueID, MY_HWID_PADDING_BYTE, sizeof(unique_id_t));
+	// no unique ID for non-PB AVR, use HW specifics for diversification
+	*((uint8_t *)uniqueID) = SIGNATURE_2;
+	*((uint8_t *)uniqueID + 1) = SIGNATURE_1;
+	*((uint8_t *)uniqueID + 2) = SIGNATURE_0;
+	*((uint8_t *)uniqueID + 3) = SIGROW.DEVICEID0;
+	*((uint8_t *)uniqueID + 4) = SIGROW.DEVICEID1;
+	*((uint8_t *)uniqueID + 5) = SIGROW.DEVICEID2;
+
+	// SIGROW.SERNUM[9:0].
+	for(uint8_t idx = 0; idx < 10; idx++) {
+		*((uint8_t *)uniqueID + 6 + idx) = *(&SIGROW.SERNUM0 + idx);
+	}
+
+
+	return true; // unique ID returned
+}
+
+uint16_t hwCPUVoltage(void)
+{
+	analogReference(INTERNAL1V024);
+
+	for (uint8_t i = 0; i < 10; i++) {
+		analogRead(ADC_VDDDIV10);
+	}
+
+	uint16_t adc = analogRead(ADC_VDDDIV10);
+	uint16_t voltage = adc * (uint16_t) 10.24; // value in mV
+
+	return voltage; // value in mV
+}
+
+uint16_t hwCPUFrequency(void)
+{
+#ifdef F_CPU
+	return F_CPU / 100000UL;
+#else
+	cli();
+	const uint8_t rtcPitCtrlA = RTC.PITCTRLA;
+	const uint8_t rtcPitIntCtrl = RTC.PITINTCTRL;
+	const uint8_t rtcPitPitIntFlags = RTC.PITINTFLAGS;
+
+	const uint8_t tcaEvCtrl = TCA0.SINGLE.EVCTRL;
+	const uint8_t tcaCtrlA = TCA0.SINGLE.CTRLA;
+
+
+	// setup timer1
+	TCA0.SINGLE.CNT = 0x0; /* Count: 0x0 */
+
+	TCA0.SINGLE.EVCTRL = 0 << TCA_SINGLE_CNTAEI_bp           /* Count on Event Input A: disabled */
+	                     | 0 << TCA_SINGLE_CNTBEI_bp            /* Count on Event Input B: disabled */
+	                     | TCA_SINGLE_EVACTA_CNT_POSEDGE_gc     /* Count on positive edge event */
+	                     | TCA_SINGLE_EVACTB_UPDOWN_gc;         /* Count on prescaled clock. Event controls count direction.
+																															Up-count when event line is 0, down-count when event line
+																															is 1. */
+	TCA0.SINGLE.CTRLA = TCA_SINGLE_CLKSEL_DIV1024_gc         /* System Clock */
+	                    | 1 << TCA_SINGLE_ENABLE_bp            /* Module Enable: enabled */
+	                    | 0 << TCA_SINGLE_RUNSTDBY_bp;         /* Run Standby: disabled */
+
+	// set pit
+	while (RTC.PITSTATUS >
+	        0) {}                             /* Wait for all register to be synchronized */
+	RTC.PITCTRLA = RTC_PERIOD_CYC16384_gc                    /* RTC Clock Cycles 16384 - 500ms */
+	               | 1 << RTC_PITEN_bp;                      /* Enable: enabled */
+	RTC.PITINTCTRL = 1 << RTC_PI_bp;                         /* Periodic Interrupt: enabled */
+
+	// wait for pit to interrupt
+	while (bit_is_clear(RTC.PITINTFLAGS, RTC_PI_bp)) {};
+
+	TCA0.SINGLE.CTRLA = 0;                                   /* Enable: disabled */
+
+	RTC.PITCTRLA &= ~(1 << RTC_PITEN_bp);                    /* Enable: disabled */
+
+	sei();
+
+	uint16_t count = TCA0.SINGLE.CNT;
```

Comment:
**Reporter**: Cppcheck
**Rule**: unreadVariable
**Severity**: INFO
**File**: hal/architecture/AVR/MyHwMegaAVR.cpp L395

Variable &#39;count&#39; is assigned a value that is never used.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a658707407>*